pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -511,11 +496,11 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 					sort.Strings(routes)
 					sort.Strings(con.Routes)
 
-					if reflect.DeepEqual(con.Routes, routes) {
+					if reflect.DeepEqual(con.Routes, routes) || len(routes) == 0 {
```

Comment:
no need too

Reply:
The 'ack' handling is pretty strange and may change with 'on-demand' - better to be safe. 
My understanding is that 'on-demand' will use an ack that has only status code, doesn't include full routes. But I can move this to separate PR.

Reply:
I see it was moved up ( had to resolve some conflicts when merging - I think that's the old code.)